storage_service/locations/models/package.py

```diff
@@ -752,6 +754,104 @@ def backlog_transfer(self, origin_location, origin_path):
         self.status = Package.UPLOADED
         self.save()
 
+    def index_file_data_from_aip_mets(self):
+        """
+        Attempts to read an Archivematica AIP METS file inside this
+        package, then uses the retrieved metadata to generate one entry in the
+        File table in the database for each file inside the package.
+
+        :raises StorageException: if the transfer METS cannot be found,
+            or if required elements are missing.
+        """
+        prefix = self.full_path
+
+        import metsrw
+
+        aip_dir_name = os.path.basename(os.path.splitext(self.full_path)[0])
+        temp_dir = tempfile.mkdtemp()
+        relative_path = os.path.join(aip_dir_name, "data", "METS." + self.uuid + ".xml")
+
+        self.extract_file(relative_path, temp_dir)
+
+        mw = metsrw.mets.METSWriter()
```

Comment:
Note: This will need to be updated if/when artefactual-labs/mets-reader-writer#9 gets merged.